MCprep_addon/spawner/mcmodel.py

```diff
@@ -0,0 +1,256 @@
+from bpy_extras.object_utils import AddObjectHelper, object_data_add
+from bpy_extras.io_utils import ImportHelper
+
+# addon imports
+from .. import conf
+from .. import util
+
+def rotate_around(d,pos,origin,axis='z',offset=[8,0,8],scale=[0.063,0.063,0.063]):
+    r=-radians(d)
+    axis_i = ord(axis)-120 # 'x'=0, 'y'=1, 'z'=2
+    a=pos[(1+axis_i)%3]
+    b=pos[(2+axis_i)%3]
+    c=pos[(3+axis_i)%3]
+    m=origin[(1+axis_i)%3]
+    n=origin[(2+axis_i)%3]
+    # this equation rotates the verticies around the origin point
+    new_pos=[0,0,0]
+    new_pos[(1+axis_i)%3]=cos(r)*(a-m)+(b-n)*sin(r)+m
+    new_pos[(2+axis_i)%3]=-sin(r)*(a-m)+cos(r)*(b-n)+n
+    new_pos[(3+axis_i)%3]=c
+    ''' offset and scale are applied to the vertices in the return
+    the default offset is what makes sure the block/item is centered
+    the default scale will match the scale of the blocks from mineways, or 1 gridspace'''
+    return Vector((
+        -(new_pos[0]-offset[0])*scale[0],
+        (new_pos[2]-offset[2])*scale[2],
+        (new_pos[1]-offset[1])*scale[1]
+        ))
+
+def add_element(elm_from=[0,0,0],elm_to=[16,16,16],rot_origin=[8,8,8],rot_axis='y',rot_angle=0):
+    # this function calculates and defines the verts, edge, and faces that will be created
+    verts = [
+        rotate_around(rot_angle,[elm_from[0], elm_to[1], elm_from[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_to[0], elm_to[1], elm_from[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_to[0], elm_from[1], elm_from[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_from[0], elm_from[1], elm_from[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_from[0], elm_to[1], elm_to[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_to[0], elm_to[1], elm_to[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_to[0], elm_from[1], elm_to[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_from[0], elm_from[1], elm_to[2]],rot_origin,rot_axis),
+    ]
+
+    edges = []
+    faces = [[0, 1, 2, 3],[5, 4, 7, 6],[1, 0, 4, 5],[7, 6, 2, 3],[4, 0, 3, 7],[1, 5, 6, 2]]
+    
+    return [verts,edges,faces]
+
+def add_material(name="material", pth=""): # creates a simple material with an image texture from pth
+    conf.log(name+": "+pth)
+    mat = bpy.data.materials.new(name=name)
+    mat.blend_method = 'CLIP'
+    mat.shadow_method = 'CLIP'
+    mat.use_nodes = True
+    matnodes = mat.node_tree.nodes
+
+    tex = matnodes.new('ShaderNodeTexImage')
+    if os.path.isfile(pth):
+        img = bpy.data.images.load(pth, check_existing=True)
+        tex.image=img
+    tex.interpolation='Closest'
+
+    shader = matnodes['Principled BSDF']
+    mat.node_tree.links.new(shader.inputs['Base Color'], tex.outputs['Color'])
+    mat.node_tree.links.new(shader.inputs['Alpha'], tex.outputs['Alpha'])
+
+    return mat
+
+def locate_image(textures,img,model_filepath):
+
+    context=bpy.context
```

Comment:
Is it possible to pass context into the function directly, instead of calling for bpy.context here?

Same applies to other cases where you immediately define context = bpy.context